packages/core/data-transfer/src/strapi/providers/utils.ts

```diff
@@ -34,26 +34,40 @@ export const createDispatcher = (ws: WebSocket) => {
     return new Promise<U | null>((resolve, reject) => {
       const uuid = randomUUID();
       const payload = { ...message, uuid };
+      let numberOfTimesMessageWasSent = 0;
 
       if (options.attachTransfer) {
         Object.assign(payload, { transferID: state.transfer?.id });
       }
 
       const stringifiedPayload = JSON.stringify(payload);
-
       ws.send(stringifiedPayload, (error) => {
         if (error) {
           reject(error);
         }
       });
 
+      const sendPeriodically = () => {
+        if (numberOfTimesMessageWasSent < 5) {
+          numberOfTimesMessageWasSent += 1;
+          ws.send(stringifiedPayload, (error) => {
+            if (error) {
+              reject(error);
+            }
+          });
+        } else {
+          reject(new ProviderError('error', 'Request timed out'));
+        }
+      };
+      const interval = setInterval(sendPeriodically, 30000);
```

Comment:
Can we set this in a variable (ideally a config option passed in when the connection is created) so we know what it's referring to?